tests/models/test_identifier.py

```diff
@@ -179,131 +214,153 @@ def test_parse_urns(self):
+        isbn_identifier, _ = Identifier.parse_urn(db_session, isbn_urn)
         assert Identifier.ISBN == isbn_identifier.type
         assert "9781449358068" == isbn_identifier.identifier
 
         isbn_urn = "urn:isbn:9781449358068"
-        isbn_identifier2, ignore = Identifier.parse_urn(self._db, isbn_urn)
+        isbn_identifier2, _ = Identifier.parse_urn(db_session, isbn_urn)
         assert isbn_identifier2 == isbn_identifier
 
         # We can parse ordinary http: or https: URLs into URI
         # identifiers.
-        http_identifier, ignore = Identifier.parse_urn(
-            self._db, "http://example.com")
+        http_identifier, _ = Identifier.parse_urn(db_session, "http://example.com")
         assert Identifier.URI == http_identifier.type
         assert "http://example.com" == http_identifier.identifier
 
-        https_identifier, ignore = Identifier.parse_urn(
-            self._db, "https://example.com")
+        https_identifier, _ = Identifier.parse_urn(db_session, "https://example.com")
         assert Identifier.URI == https_identifier.type
         assert "https://example.com" == https_identifier.identifier
 
         # We can parse UUIDs.
-        uuid_identifier, ignore = Identifier.parse_urn(
-            self._db, "urn:uuid:04377e87-ab69-41c8-a2a4-812d55dc0952")
+        uuid_identifier, _ = Identifier.parse_urn(db_session, "urn:uuid:04377e87-ab69-41c8-a2a4-812d55dc0952")
         assert Identifier.URI == uuid_identifier.type
         assert "urn:uuid:04377e87-ab69-41c8-a2a4-812d55dc0952" == uuid_identifier.identifier
 
         # A URN we can't handle raises an exception.
         ftp_urn = "ftp://example.com"
-        pytest.raises(ValueError, Identifier.parse_urn, self._db, ftp_urn)
+        pytest.raises(ValueError, Identifier.parse_urn, db_session, ftp_urn)
 
         # An invalid ISBN raises an exception.
-        pytest.raises(ValueError, Identifier.parse_urn, self._db, "urn:isbn:notanisbn")
+        pytest.raises(ValueError, Identifier.parse_urn, db_session, "urn:isbn:notanisbn")
 
         # Pass in None and you get None.
-        assert None == Identifier.parse_urn(self._db, None)
+        assert Identifier.parse_urn(db_session, None) is None
 
-    def parse_urn_must_support_license_pools(self):
+    def test_parse_urn_must_support_license_pools(self, db_session):
+        """
+        GIVEN: An ISBN URN
+        WHEN:  Parsing the ISBN URN
+        THEN:  An Identifier.UnresolvableIdentifierException is raised
+        """
         # We have no way of associating ISBNs with license pools.
         # If we try to parse an ISBN URN in a context that only accepts
         # URNs that can have associated license pools, we get an exception.
         isbn_urn = "urn:isbn:1449358063"
+
+        # Delete all DataSources to force the exception
+        rows = db_session.query(DataSource).all()
+        for row in rows:
+            db_session.delete(row)
+        db_session.flush()
```

Comment:
This shouldn't be necessary. The only way I can see this making sense is if some other piece of code is establishing a DataSource that provides LicensePools for ISBN, and that DataSource isn't being rolled back when the test ended.